presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -244,32 +223,61 @@ public boolean equals(Object o)
                 return false;
             }
 
-            if (Symbol.class.equals(o.getClass())) {
-                Symbol other = (Symbol) o;
-                String name = other.getName();
-                int endIndex = name.lastIndexOf('_');
-                String otherStem = endIndex == -1 ? name : name.substring(0, endIndex);
-                return stem.equals(otherStem);
-            }
-
-            if (!SymbolStem.class.equals(o.getClass())) {
+            if (!clazz.equals(o.getClass())) {
                 return false;
             }
 
-            SymbolStem other = (SymbolStem) o;
-            return stem.equals(other.stem);
+            T other = clazz.cast(o);
+            String name = getName.apply(other);
+            int endIndex = name.lastIndexOf('_');
+            String otherStem = endIndex == -1 ? name : name.substring(0, endIndex);
+            return getStem().equals(otherStem);
+        }
+    }
+
+    public static class SymbolStem extends Symbol
+        implements Stem
+    {
+        private final String stem;
+
+        public SymbolStem(String stem)
+        {
+            super(stem + "_*");
+            this.stem = requireNonNull(stem);
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            return this.stemEquals(o, Symbol.class, Symbol::getName);
```

Comment:
The compiler does not like it without it. It seems to think that the assignment is to the local variable stem rather than the member variable.